packages/astro/components/ViewTransitions.astro

```diff
@@ -220,6 +224,12 @@ const { fallback = 'animate' } = Astro.props as Props;
 		document.head.append(link);
 	}
 
+	function isSamePage(location, link) {
+		const linkURL = new URL(link.href, location.href);
+		linkURL.hash = location.hash;
+		return linkURL.href.replace(/#$/, '') === location.href.replace(/#$/, '');
```

Comment:
Can you explain what this is doing? It's not obvious to me. Can we not just check if `location.pathname === link.pathname`?

Reply:
Ha :smile: When I started, it seemed like the natural way to check if two URLs are "equal modulo the fragment ID". It got more complicated when I noticed a problem with hashes at the end.

But I probably mostly avoided worrying about other URL properties like searchParams.

I only have SSG experience with Astro.  
I suppose there are also SSR examples with fixed pathnames, where people expect different searchParams to show the transition to different pages?

Maybe we should aim for a simple, generic solution and give users an additional link property for manual control.

pathname === pathname is fine with me. I will fix it that way.

Maybe you already have ideas on how to override link behavior to force transitions for same pathnames. I'll leave that to a later change :-)